tests/contest/contest/src/tests/cgroups/relative_network.rs

```diff
@@ -0,0 +1,94 @@
+use std::path::Path;
+
+use anyhow::{Context, Result};
+use oci_spec::runtime::{
+    LinuxBuilder, LinuxInterfacePriorityBuilder, LinuxNetworkBuilder, LinuxResourcesBuilder, Spec,
+    SpecBuilder,
+};
+use pnet_datalink::interfaces;
+use test_framework::{test_result, ConditionalTest, TestGroup, TestResult};
+
+use crate::utils::test_outside_container;
+use crate::utils::test_utils::check_container_created;
+
+fn create_spec(cgroup_name: &str, class_id: u32, prio: u32, if_name: &str) -> Result<Spec> {
+    // Create the Linux Spec
+    let linux_spec = LinuxBuilder::default()
+        .cgroups_path(Path::new("testdir/runtime-test/container").join(cgroup_name))
+        .resources(
+            LinuxResourcesBuilder::default()
+                .network(
+                    LinuxNetworkBuilder::default()
+                        .class_id(class_id)
+                        .priorities(vec![LinuxInterfacePriorityBuilder::default()
+                            .name(if_name)
+                            .priority(prio)
+                            .build()
+                            .context("failed to build network interface priority spec")?])
+                        .build()
+                        .context("failed to build network spec")?,
+                )
+                .build()
+                .context("failed to build resource spec")?,
+        )
+        .build()
+        .context("failed to build linux spec")?;
+
+    // Create the top level Spec
+    let spec = SpecBuilder::default()
+        .linux(linux_spec)
+        .build()
+        .context("failed to build spec")?;
+
+    Ok(spec)
+}
+
+// Gets the loopback interface if it exists
+fn get_loopback_interface() -> Option<String> {
+    let interfaces = interfaces();
+    let lo_if_name = interfaces.first().map(|iface| &iface.name)?;
+
+    Some(lo_if_name.to_string())
+}
+
+fn test_relative_network_cgroups() -> TestResult {
+    let cgroup_name = "test_relative_network_cgroups";
+
+    let id = 255;
+    let prio = 10;
+    let if_name = "lo";
+    let spec = test_result!(create_spec(cgroup_name, id, prio, if_name));
+
+    test_outside_container(spec, &|data| {
+        test_result!(check_container_created(&data));
+        TestResult::Passed
+    })
```

Comment:
Hey, here along with checking if the container is created, we also need validation for the created network cgroup resources - In the [original test](https://github.com/opencontainers/runtime-tools/blob/master/validation/linux_cgroups_relative_network/linux_cgroups_relative_network.go#L24) we call [this](https://github.com/opencontainers/runtime-tools/blob/master/validation/util/linux_resources_network.go#L12) function which does the validation, so need that here as well.

Reply:
Thank you.
Added validation for the created network cgroup resources.

Reply:
Hey, I don't think it is fixed yet. Let me clarify in case there is any confusion -
1. In the original go test, at line https://github.com/opencontainers/runtime-tools/blob/master/validation/linux_cgroups_relative_network/linux_cgroups_relative_network.go#L24C1-L24C77, in the `test_outside_container` , they are passing `util.ValidateLinuxResourcesNetwork` function, which will do the validation that ok, the runtime has actually setup the relative network correctly.
2. The `util.ValidateLinuxResourcesNetwork` function defined at https://github.com/opencontainers/runtime-tools/blob/master/validation/util/linux_resources_network.go#L12 does the checking and validation of relative network cgroup.
3. The change you did in the last commit you pushed is actually almost a no-op. The original way of just calling the `test_outside_container` was correct, but also needs the cgroup checking logic as mentioned above.

Reply:
I added a new `validate_network()` and changed the program to validate net_cls.classid and net_prio.ifpriomap.
However, [network.rs](https://github.com/youki-dev/youki/blob/main/tests/contest/contest/src/tests/cgroups/network.rs) only verifies `check_container_created(&data)`.
Am I understanding this wrong?